handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -89,9 +91,18 @@ public SniHandler(DomainNameMapping<? extends SslContext> mapping) {
      *
      * @param mapping the mapping of domain name to {@link SslContext}
      */
-    @SuppressWarnings("unchecked")
     public SniHandler(AsyncMapping<? super String, ? extends SslContext> mapping) {
-        this.mapping = (AsyncMapping<String, SslContext>) ObjectUtil.checkNotNull(mapping, "mapping");
+        this (new AsyncMappingAdapter(mapping));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)